hazelcast/src/main/java/com/hazelcast/map/impl/MapKeyLoader.java

```diff
@@ -439,4 +451,14 @@ public String toString() {
     public void onKeyLoad(ExecutionCallback<Boolean> callback) {
         loadFinished.andThen(callback, execService.getExecutor(MAP_LOAD_ALL_KEYS_EXECUTOR));
     }
+
+    public void promoteToLoadedOnMigration() {
+        state.next(State.LOADING);
+        state.next(State.LOADED);
```

Comment:
The state machine cannot skip states - it's like a graph with connections. I don't want to modify the graph for this particular case, since it may be a smell in different cases.